### PR TITLE
FileDropZone: Add reset button during error state

### LINKED ISSUE #851: File Upload Dropzone - Not able to reset error state
Once an error state is achieved looks like there is no way to attempt the upload again, trying to drop a valid file is not clearing the error message.

1. Go to [File upload drop zone](https://modus-web-components.trimble.com/?path=/docs/components-file-dropzone--default)
2. Upload multiple files and appears an error message
3. Attempt upload again with a valid file, the error message doesn't go away, and the file used in the upload is opening in the browser.

**Expected behavior**
The error message should be cleared when the upload is valid.

chrome_ZwIPPkwfE2

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

- Added Reset button in error state

References 
Fixes #851 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-// eslint-disable-next-line
 import { Component, h, Prop, State, Event, EventEmitter, Method, Listen, Host } from '@stencil/core';
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx:1]: @ElishaSamPeterPrabhu please revert this change

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx [hunk 2]
```diff
@@ -140,6 +138,14 @@ export class ModusFileDropzone {
     this.fileInput.click();
   };
 
+  reset = (): void => {
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx:141]: @ElishaSamPeterPrabhu please add tests for this new behavior

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.tsx]: Reverted

File: stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.spec.ts [hunk 3]
```diff
@@ -86,4 +86,163 @@ describe('modus-file-dropzone', () => {
       </modus-file-dropzone>
     `);
   });
+
+  it('renders with maxFileCount error is triggered', async () => {
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.spec.ts:90]: @ElishaSamPeterPrabhu Write E2E for behaviors instead spec.

Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-file-dropzone/modus-file-dropzone.spec.ts:90]: Added e2e tests

**Review by prashanth-offcl — APPROVED** [review]
